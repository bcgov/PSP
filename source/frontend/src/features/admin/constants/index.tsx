import { ColumnWithProps } from 'components/Table';
import { IOrganization, IOrganizationRecord } from 'interfaces';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';

export const columnDefinitions: ColumnWithProps<IOrganizationRecord>[] = [
  {
    Header: 'Organization name',
    accessor: 'name',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Short name',
    accessor: 'code',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Description',
    accessor: 'description',
    align: 'left',
    clickable: true,
  },
  {
    Header: 'Parent Organization',
    accessor: 'parent',
    align: 'left',
    clickable: true,
    Cell: (props: CellProps<IOrganization>) => {
      return (
        <Link to={`/admin/organization/${props.row.original.parentId}`}>
          {props.row.original.parent}
        </Link>
      );
    },
  },
];
