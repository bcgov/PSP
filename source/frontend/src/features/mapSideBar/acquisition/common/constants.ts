export const cannotEditMessage =
  'This section of the file is in a non-editable state. Contact a System administrator to change it to an editable state';
