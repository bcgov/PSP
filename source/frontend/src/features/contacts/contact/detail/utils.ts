import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
