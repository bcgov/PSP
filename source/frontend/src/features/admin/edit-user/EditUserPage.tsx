import { H1 } from 'components/common/styles';
import { Col, Row } from 'react-bootstrap';
import styled from 'styled-components';

import EditUserContainer from './EditUserContainer';

interface IEditUserPageProps {
  userKey: string;
  match?: any;
}

const EditUserPage = (props: IEditUserPageProps) => {
  const userId = props?.match?.params?.key || props.userKey;

  return (
    <StyledContainer className="EditUserPage">
      <Row>
        <Col md={7}>
          <H1>User Information</H1>
        </Col>
      </Row>
      <Row>
        <Col md={7}>
          <EditUserContainer userId={userId} />
        </Col>
      </Row>
    </StyledContainer>
  );
};

const StyledContainer = styled.div`
  width: 100%;
  overflow-y: auto;
  padding: 3rem;
  > .row {
    justify-content: center;
  }
  form {
    text-align: left;
  }
`;

export default EditUserPage;
