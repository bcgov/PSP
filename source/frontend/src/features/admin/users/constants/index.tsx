import { ReactComponent as Active } from 'assets/images/active.svg';
import { ReactComponent as Inactive } from 'assets/images/inactive.svg';
import { ColumnWithProps } from 'components/Table';
import { DateTimeCell } from 'components/Table/DateCell';
import { Link } from 'react-router-dom';
import { CellProps } from 'react-table';
import { stringToFragment } from 'utils';

import { RowActions } from '../components/RowActions';
import { FormUser } from '../models';

export const getUserColumns = (refresh: () => void): ColumnWithProps<FormUser>[] => [
  {
    Header: 'Active',
    accessor: 'isDisabled',
    align: 'left',
    clickable: true,
    sortable: true,
    width: 60,
    Cell: (props: CellProps<FormUser>) =>
      props.row.original.isDisabled ? <Inactive /> : <Active />,
  },
  {
    Header: 'IDIR/BCeID',
    accessor: 'businessIdentifierValue',
    align: 'left',
    minWidth: 150,
    Cell: (props: CellProps<FormUser>) => {
      return (
        <Link to={`/admin/user/${props.row.original.id}`}>
          {props.row.original.businessIdentifierValue}
        </Link>
      );
    },
    sortable: true,
  },
  {
    Header: 'First name',
    accessor: 'firstName',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Last name',
    accessor: 'surname',
    align: 'left',
    clickable: true,
    sortable: true,
  },
  {
    Header: 'Email',
    accessor: 'email',
    align: 'left',
    clickable: true,
    sortable: true,
    minWidth: 200,
  },
  {
    Header: 'Position',
    accessor: 'position',
    align: 'left',
    clickable: true,
    sortable: true,
  },

  {
    Header: 'Roles',
    accessor: 'roles',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const rolesString = props.row?.original?.roles?.map(userRole => userRole?.name);
      return stringToFragment(rolesString?.join(', '));
    },
  },
  {
    Header: 'MoTI region(s)',
    accessor: 'regions',
    align: 'left',
    clickable: true,
    minWidth: 200,
    Cell: (props: CellProps<FormUser>) => {
      const regionsString = props.row?.original?.regions?.map(
        regionUser => regionUser?.description,
      );
      return stringToFragment(regionsString?.join(', '));
    },
  },
  {
    Header: 'Last login',
    accessor: 'lastLogin',
    align: 'left',
    clickable: true,
    minWidth: 100,
    Cell: DateTimeCell,
  },
  {
    Header: ' ',
    Cell: (props: CellProps<FormUser>) => {
      return <RowActions {...props} refresh={refresh} />;
    },
    width: 75,
  },
];
