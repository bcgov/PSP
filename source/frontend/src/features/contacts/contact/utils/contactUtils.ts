import { CountryCodes } from 'constants/index';
import { AddressField } from 'features/contacts/interfaces';
import { validateYupSchema, yupToFormErrors } from 'formik';
import { IEditableOrganizationForm, IEditablePersonForm } from 'interfaces/editable-contact';
import isEmpty from 'lodash/isEmpty';

import {
  hasAddress,
  hasEmail,
  hasPhoneNumber,
  OrganizationValidationSchema,
  PersonValidationSchema,
} from '../create/validation';
import { sortAddresses } from '../detail/utils';
import { IContactAddress } from './../../../../interfaces/IContact';

export const onValidateOrganization = (
  values: IEditableOrganizationForm,
  otherCountryId?: string,
) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, OrganizationValidationSchema, true, {
      otherCountry: otherCountryId,
    });
    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const onValidatePerson = (values: IEditablePersonForm, otherCountryId?: string) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, PersonValidationSchema, true, { otherCountry: otherCountryId });

    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const toAddressFields = (addresses: IContactAddress[]) => {
  return addresses
    .sort(sortAddresses)
    .reduce((accumulator: AddressField[], value: IContactAddress) => {
      accumulator.push({
        label: value.addressType.description || '',
        streetAddress1: value.streetAddress1,
        streetAddress2: value.streetAddress2,
        streetAddress3: value.streetAddress3,
        municipalityAndProvince: [
          value.municipality !== undefined ? value.municipality : '',
          value.province?.provinceStateCode ?? '',
        ]
          .filter(x => !isEmpty(x))
          .join(' '),
        country:
          value.country?.countryCode === CountryCodes.Other
            ? value.countryOther ?? ''
            : value.country?.description,
        postal: value.postal,
      });
      return accumulator;
    }, []);
};
