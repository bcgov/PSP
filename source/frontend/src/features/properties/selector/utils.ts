import { IMapProperty } from 'features/properties/selector/models';
import { compact } from 'lodash';
import { Api_PropertyFile } from 'models/api/PropertyFile';
import { formatApiAddress, pidFormatter } from 'utils';

import { Api_Geometry, Api_Property } from './../../../models/api/Property';

export enum NameSourceType {
  PID = 'PID',
  PIN = 'PIN',
  PLAN = 'Plan #',
  LOCATION = 'Location',
  NAME = 'Descriptive Name',
  NONE = 'n/a',
  ADDRESS = 'Address',
}

interface PropertyName {
  label: NameSourceType;
  value: string;
}

export const getPropertyName = (property: IMapProperty): PropertyName => {
  if (property.pid !== undefined && property.pid?.toString().length > 0 && property.pid !== '0') {
    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
  } else if (
    property.pin !== undefined &&
    property.pin?.toString()?.length > 0 &&
    property.pin !== '0'
  ) {
    return { label: NameSourceType.PIN, value: property.pin.toString() };
  } else if (property.planNumber !== undefined) {
    return { label: NameSourceType.PLAN, value: property.planNumber };
  } else if (property.latitude !== undefined && property.longitude !== undefined) {
    return {
      label: NameSourceType.LOCATION,
      value: compact([property.longitude?.toFixed(6), property.latitude?.toFixed(6)]).join(', '),
    };
  } else if (property.address !== undefined) {
    return {
      label: NameSourceType.ADDRESS,
      value: property.address,
    };
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getPrettyLatLng = (location?: Api_Geometry) =>
  compact([
    location?.coordinate?.x?.toFixed(6) ?? 0,
    location?.coordinate?.y?.toFixed(6) ?? 0,
  ]).join(', ');

export const getFilePropertyName = (
  fileProperty?: Api_PropertyFile,
  skipName: boolean = false,
): PropertyName => {
  if (fileProperty === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  if (
    fileProperty.propertyName !== undefined &&
    fileProperty.propertyName !== '' &&
    skipName === false
  ) {
    return { label: NameSourceType.NAME, value: fileProperty.propertyName };
  } else if (fileProperty.property !== undefined) {
    const property = fileProperty.property;
    return getApiPropertyName(property);
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getApiPropertyName = (property?: Api_Property): PropertyName => {
  if (property === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  let mapProperty: IMapProperty = {
    pin: property.pin?.toString(),
    pid: property.pid?.toString(),
    latitude: property.latitude,
    longitude: property.longitude,
    planNumber: property.planNumber,
    address: property.address !== undefined ? formatApiAddress(property.address) : undefined,
  };
  return getPropertyName(mapProperty);
};
