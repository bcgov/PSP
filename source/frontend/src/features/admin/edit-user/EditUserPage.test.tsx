import { useKeycloak } from '@react-keycloak/web';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import * as API from 'constants/API';
import { createMemoryHistory } from 'history';
import { getUserMock } from 'mocks/userMock';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ILookupCode, lookupCodesSlice } from 'store/slices/lookupCodes';
import { act, cleanup, render, waitForElementToBeRemoved } from 'utils/test-utils';

import EditUserPage from './EditUserPage';

jest.mock('@react-keycloak/web');
(useKeycloak as jest.Mock).mockReturnValue({
  keycloak: {
    userInfo: {
      organizations: [1],
      roles: [],
    },
    subject: 'test',
  },
});

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();

const lCodes = {
  lookupCodes: [
    { name: 'roleVal', id: 1, isDisabled: false, type: API.ROLE_TYPES },
    { name: 'disabledRole', id: 2, isDisabled: true, type: API.ROLE_TYPES },
  ] as ILookupCode[],
};

const store = mockStore({
  [lookupCodesSlice.name]: lCodes,
});

const mockAxios = new MockAdapter(axios);

const renderEditUserPage = () =>
  render(
    <Provider store={store}>
      <Router history={history}>
        <EditUserPage userKey="TEST-ID" />,
      </Router>
    </Provider>,
  );

describe('Edit user page', () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });
  beforeEach(() => {
    mockAxios.reset();
    mockAxios.onAny().reply(200, getUserMock());
  });

  it('EditUserPage renders', async () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <Router history={history}>
          <EditUserPage userKey="TEST-ID" />,
        </Router>
      </Provider>,
    );
    await waitForElementToBeRemoved(getByTestId('filter-backdrop-loading'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains role options from lookup code + please select disabled option', async () => {
    const { getAllByText, getByTestId, findByDisplayValue } = renderEditUserPage();
    await findByDisplayValue('pos');
    expect(getAllByText(/roleVal/i)).toHaveLength(1);
    expect(getByTestId('isDisabled').getAttribute('value')).toEqual('false');
  });

  it('Does not display disabled roles', async () => {
    const { queryByText, findByDisplayValue } = renderEditUserPage();
    await findByDisplayValue('pos');
    expect(queryByText('disabledRole')).toBeNull();
  });

  describe('appropriate fields are autofilled', () => {
    it('autofills  email, businessIdentifier, first and last name', async () => {
      const { getByTestId, findByDisplayValue } = renderEditUserPage();
      await findByDisplayValue('pos');

      expect(getByTestId('email').getAttribute('value')).toEqual('devin.smith@gov.bc.ca');
      expect(getByTestId('businessIdentifier').getAttribute('value')).toEqual('desmith@idir');
      expect(getByTestId('firstName').getAttribute('value')).toEqual('Devin');
      expect(getByTestId('surname').getAttribute('value')).toEqual('Smith');
      expect(getByTestId('position').getAttribute('value')).toEqual('pos');
    });
  });

  describe('when the user edit form is submitted', () => {
    it('displays a loading toast', async () => {
      const { getByText, findByText, findByDisplayValue } = renderEditUserPage();
      const saveButton = getByText('Save');
      mockAxios.onGet().reply(200, getUserMock());
      await findByDisplayValue('pos');
      act(() => {
        saveButton.click();
      });
      await findByText('Updating User...');
    });

    it('displays a success toast if the request passes', async () => {
      const { getByText, findByText, findByDisplayValue } = renderEditUserPage();
      const saveButton = getByText('Save');
      mockAxios.onGet().reply(200, getUserMock());
      mockAxios.onPut().reply(200, {});
      await findByDisplayValue('pos');
      act(() => {
        saveButton.click();
      });
      await findByText('User updated');
    });

    it('displays an error toast if the request fails', async () => {
      const { getByText, findByText, findByDisplayValue } = renderEditUserPage();
      const saveButton = getByText('Save');
      mockAxios.onGet().replyOnce(200, getUserMock());
      mockAxios.onPut().reply(500, {});
      await findByDisplayValue('pos');
      act(() => {
        saveButton.click();
      });
      await findByText('Failed to update User');
    });
  });
});
