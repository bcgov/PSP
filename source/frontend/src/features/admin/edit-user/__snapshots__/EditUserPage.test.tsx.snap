// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit user page EditUserPage renders 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
  <div
    class="Toastify"
  />
  .c6.btn {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.4rem 1.2rem;
  min-height: 3rem;
  border: 0.2rem solid transparent;
  border-radius: 0.4rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.8rem;
  font-family: 'BCSans','Noto Sans',Verdana,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  cursor: pointer;
}

.c6.btn:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c6.btn:focus {
  outline-width: 0.4rem;
  outline-style: solid;
  outline-offset: 1px;
  box-shadow: none;
}

.c6.btn.btn-primary {
  border: none;
}

.c6.btn.btn-secondary {
  background: none;
}

.c6.btn.btn-info {
  border: none;
  background: none;
  padding-left: 0.6rem;
  padding-right: 0.6rem;
}

.c6.btn.btn-info:hover,
.c6.btn.btn-info:active,
.c6.btn.btn-info:focus {
  background: none;
}

.c6.btn.btn-light {
  border: none;
}

.c6.btn.btn-dark {
  border: none;
}

.c6.btn.btn-link {
  font-size: 1.6rem;
  font-weight: 400;
  background: none;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 2.5rem;
  line-height: 3rem;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-letter-spacing: unset;
  -moz-letter-spacing: unset;
  -ms-letter-spacing: unset;
  letter-spacing: unset;
  text-align: left;
  padding: 0;
}

.c6.btn.btn-link:hover,
.c6.btn.btn-link:active,
.c6.btn.btn-link:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  border: none;
  background: none;
  box-shadow: none;
  outline: none;
}

.c6.btn.btn-link:disabled,
.c6.btn.btn-link.disabled {
  background: none;
  pointer-events: none;
}

.c6.btn:disabled,
.c6.btn:disabled:hover {
  box-shadow: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.65;
}

.c6.Button .Button__icon {
  margin-right: 1.6rem;
}

.c6.Button--icon-only:focus {
  outline: none;
}

.c6.Button--icon-only .Button__icon {
  margin-right: 0;
}

.c4 {
  margin-left: 0.5rem;
  cursor: pointer;
}

.c1 {
  font-family: 'BCSans-Bold';
  font-size: 3.2rem;
  border-bottom: solid 0.5rem;
  width: 100%;
  text-align: left;
  margin-bottom: 2rem;
}

.c3.required::before {
  content: '*';
  position: absolute;
  top: 0.75rem;
  left: 0rem;
}

.c2 {
  font-weight: bold;
}

.c5 {
  position: absolute;
  top: 1rem;
  right: -0.5rem;
}

.c0 {
  width: 100%;
  overflow-y: auto;
  padding: 3rem;
}

.c0 > .row {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 form {
  text-align: left;
}

<div
    class="c0 EditUserPage"
  >
    <div
      class="row"
    >
      <div
        class="col-md-7"
      >
        <h1
          class="c1"
        >
          User Information
        </h1>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-7"
      >
        <form
          class="userInfo"
          novalidate=""
        >
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                IDIR/BCeID:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="businessIdentifier"
                  id="input-businessIdentifier"
                  name="businessIdentifier"
                  readonly=""
                  title="desmith@idir"
                  type="text"
                  value="desmith@idir"
                />
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                First name:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="firstName"
                  id="input-firstName"
                  name="firstName"
                  placeholder="Devin"
                  title="Devin"
                  type="text"
                  value="Devin"
                />
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Last name:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="surname"
                  id="input-surname"
                  name="surname"
                  placeholder="Smith"
                  title="Smith"
                  type="text"
                  value="Smith"
                />
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Email:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="email"
                  id="input-email"
                  name="email"
                  placeholder="devin.smith@gov.bc.ca"
                  title="devin.smith@gov.bc.ca"
                  type="email"
                  value="devin.smith@gov.bc.ca"
                />
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Position:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <input
                  class="form-control"
                  data-testid="position"
                  id="input-position"
                  name="position"
                  placeholder="e.g) Director, Real Estate and Stakeholder Engagement"
                  title="pos"
                  type="text"
                  value="pos"
                />
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Role(s):
              </label>
            </div>
            <div
              class="c3 required text-left col"
            >
              <div
                class="form-group"
              >
                <div>
                  <div
                    class="multiselect-container multiSelectContainer  "
                    id="multiselect-roles"
                    style="width: auto; color: black;"
                  >
                    <div
                      class="search-wrapper searchWrapper "
                      style="background: white; border: 1px solid #606060;"
                    >
                      <span
                        class="chip  false false"
                        style="background: rgb(242, 242, 242); border-radius: 4px; color: black; font-size: 16px; margin-right: 1em;"
                      >
                        Functional
                        <i
                          class="custom-close"
                        >
                          <svg
                            class="c4"
                            fill="currentColor"
                            height="16"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 512 512"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                            />
                          </svg>
                        </i>
                      </span>
                      <span
                        class="chip  false false"
                        style="background: rgb(242, 242, 242); border-radius: 4px; color: black; font-size: 16px; margin-right: 1em;"
                      >
                        System Administrator
                        <i
                          class="custom-close"
                        >
                          <svg
                            class="c4"
                            fill="currentColor"
                            height="16"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 512 512"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-83.6 290.5c4.8 4.8 4.8 12.6 0 17.4l-40.5 40.5c-4.8 4.8-12.6 4.8-17.4 0L256 313.3l-66.5 67.1c-4.8 4.8-12.6 4.8-17.4 0l-40.5-40.5c-4.8-4.8-4.8-12.6 0-17.4l67.1-66.5-67.1-66.5c-4.8-4.8-4.8-12.6 0-17.4l40.5-40.5c4.8-4.8 12.6-4.8 17.4 0l66.5 67.1 66.5-67.1c4.8-4.8 12.6-4.8 17.4 0l40.5 40.5c4.8 4.8 4.8 12.6 0 17.4L313.3 256l67.1 66.5z"
                            />
                          </svg>
                        </i>
                      </span>
                      <input
                        autocomplete="off"
                        class="searchBox "
                        id="multiselect-roles_input"
                        name="search_name_input"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="optionListContainer displayNone"
                    >
                      <ul
                        class="optionContainer"
                      >
                        <li
                          class="option  highlightOption highlight  "
                        >
                          roleVal
                        </li>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
              <svg
                class="c5 tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                MoTI Region(s):
              </label>
            </div>
            <div
              class="c3 required text-left col"
            >
              <div
                class="form-group"
              >
                <div>
                  <div
                    class="multiselect-container multiSelectContainer  "
                    id="multiselect-regions"
                    style="width: auto; color: black;"
                  >
                    <div
                      class="search-wrapper searchWrapper "
                      style="background: white; border: 1px solid #606060;"
                    >
                      <input
                        autocomplete="off"
                        class="searchBox "
                        id="multiselect-regions_input"
                        name="search_name_input"
                        placeholder=""
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="optionListContainer displayNone"
                    >
                      <ul
                        class="optionContainer"
                      >
                        <span
                          class="notFound"
                        >
                          No Options Available
                        </span>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="pb-2 row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Notes:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="input form-group"
              >
                <textarea
                  class="form-control"
                  data-testid="note"
                  id="input-note"
                  name="note"
                  placeholder="A note about this user"
                  title=""
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="d-flex row"
          >
            <div
              class="pr-0 text-left col-2"
            >
              <label
                class="c2"
              >
                Disable account?:
              </label>
            </div>
            <div
              class="c3 text-left col"
            >
              <div
                class="form-group"
              >
                <div
                  class="check-field"
                >
                  <div
                    class="form-check"
                  >
                    <input
                      class="form-check-input position-static"
                      data-testid="isDisabled"
                      id="input-isDisabled"
                      name="isDisabled"
                      type="checkbox"
                      value="false"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="cancelSave btn-toolbar"
              role="toolbar"
            >
              <button
                class="c6 Button mr-5 btn btn-secondary"
                type="button"
              >
                <div
                  class="Button__value"
                >
                  Cancel
                </div>
              </button>
              <button
                class="c6 Button mr-5 btn btn-primary"
                type="submit"
              >
                <div
                  class="Button__value"
                >
                  Save
                </div>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
  ,
</DocumentFragment>
`;
